apps/fabric-website/index.html

```diff
@@ -1,157 +1,161 @@
+        );
+
+        // Set session value if it's in array
+        if (fabricVersions.indexOf(fabricVersion) > -1) {
+          window.sessionStorage.setItem('fabricVer', fabricVersion);
+        }
 
-      var configScript = ['https://fabricweb.azureedge.net/fabric-website/manifests/' + configPrefix + '.js'];
+        var prefixVer =
+          fabricVersions.indexOf(fabricVersion) > -1 && fabricVersion !== fabricVersions.slice(-1)[0] ? 'v' + fabricVersion + '-' : '';
 
-      loadScripts(configScript, function () {
-        if (window.Flight) {
-          appPath += window.Flight.baseCDNUrl;
+        var configPrefix = isProduction ? 'fabric-website-' + prefixVer + 'prod' : 'fabric-website-' + prefixVer + 'df';
 
-          loadAppScripts(appPath);
-        }
-      });
-    }
-
-    else {
-      loadAppScripts(appPath);
-    }
-
-    // Simple synchronous script loader with callback. Adapted from https://stackoverflow.com/questions/1866717
-    function loadScripts(array, callback) {
-      var loader = function (src, handler) {
-        var script = document.createElement('script');
-        script.src = src;
-        script.onload = script.onreadystatechange = function () {
-          script.onreadystatechange = script.onload = null;
-          handler();
-        }
-        var head = document.getElementsByTagName('head')[0];
-        (head || document.body).appendChild(script);
-      };
-      (function run() {
-        if (array.length != 0) {
-          loader(array.shift(), run);
-        } else {
-          callback && callback();
-        }
-      })();
-    }
+        var configScript = ['https://fabricweb.azureedge.net/fabric-website/manifests/' + configPrefix + '.js'];
 
-    function loadAppScripts(appPath) {
-      var reactPath = '//cdnjs.cloudflare.com/ajax/libs/react/16.3.2/umd/react';
-      var reactDomPath = '//cdnjs.cloudflare.com/ajax/libs/react-dom/16.3.2/umd/react-dom';
+        loadScripts(configScript, function() {
+          if (window.Flight) {
+            appPath += window.Flight.baseCDNUrl;
 
-      if (noCompress) {
-        reactPath = reactPath + '.development.js';
-        reactDomPath = reactDomPath + '.development.js';
-        appPath += entryPointFilename + '.js';
+            loadAppScripts(appPath);
+          }
+        });
       } else {
+        loadAppScripts(appPath);
+      }
+
+      // Simple synchronous script loader with callback. Adapted from https://stackoverflow.com/questions/1866717
+      function loadScripts(array, callback) {
+        var loader = function(src, handler, failHandler) {
+          var script = document.createElement('script');
+          script.src = src;
+          script.onload = script.onreadystatechange = function() {
+            script.onreadystatechange = script.onload = null;
+            handler();
+          };
+
+          script.onerror = script.onreadystatechange = function() {
+            script.onreadystatechange = script.onerror = null;
```

Comment:
Why are we setting script.onreadystatechange then overwriting it (if I understand right how setting these event handlers works)? Also, I don't understand why we want to run the error logic just because the ready state changes.

Reply:
this is just a copy paste code of the loadscript above to null out the callbacks (a = b = null is lazy way of say a = null; b=null)